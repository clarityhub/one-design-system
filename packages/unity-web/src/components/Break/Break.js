import styled from '@emotion/styled';

import colors from '@clarityhub/unity-core/lib/colors';

const Break = styled.div`
    margin: 5rem 0;
    text-align: center;
    position: relative;

    &::after {
        content: "";
        position: absolute;
        width: .2rem;
        height: .2rem;
        border-radius: 50%;
        background: ${colors.dark.default};
        bottom: 0;
        left: 50%;
        margin-bottom: 1rem;
        transform: translateX(-50%);
        box-shadow: 1.6rem 0px 0px ${colors.dark.default}, 
              -1.6rem 0px 0px ${colors.dark.default};
    }
`;

export default Break;
