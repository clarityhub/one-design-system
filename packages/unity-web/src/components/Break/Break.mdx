---
name: Break
menu: Components
---

import { Playground, PropsTable } from 'docz';
import Break from './index';

# Break

A simple visual break between content sections.

## \<Break /\> component

```js
import Break from '@clarityhub/unity-web/lib/components/Break';
```

## Prop Types

<PropsTable of={Break} />

## Examples

<Playground>
    <Break />
</Playground>
