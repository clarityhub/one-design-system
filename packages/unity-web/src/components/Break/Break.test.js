import React from 'react';
import renderer from 'react-test-renderer';

import Break from './Break';

describe('<Break />', () => {
	it('renders a break', () => {
		const node = (
			<Break />
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});