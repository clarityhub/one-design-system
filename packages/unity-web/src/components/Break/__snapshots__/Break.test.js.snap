// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Break /> renders a break 1`] = `
.emotion-0 {
  margin: 5rem 0;
  text-align: center;
  position: relative;
}

.emotion-0::after {
  content: "";
  position: absolute;
  width: .2rem;
  height: .2rem;
  border-radius: 50%;
  background: #322439;
  bottom: 0;
  left: 50%;
  margin-bottom: 1rem;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  box-shadow: 1.6rem 0px 0px #322439,-1.6rem 0px 0px #322439;
}

<div
  className="emotion-0 emotion-1"
/>
`;
